graphql_compiler/compiler/emit_sql.py

```diff
@@ -104,6 +149,61 @@ def traverse(self, vertex_field, optional):
             onclause=(previous_alias.c[edge.from_column] == self._current_alias.c[edge.to_column]),
             isouter=optional)
 
+    def recurse(self, vertex_field, depth):
+        """Execute a Recurse Block."""
+        previous_alias = self._current_alias
+        edge = self._sql_schema_info.join_descriptors[self._current_classname][vertex_field]
+        primary_key = self._sql_schema_info.primary_keys[self._current_classname]
+        self._relocate(self._current_location.navigate_to_subpath(vertex_field))
+
+        if self._is_in_optional_scope():
+            raise NotImplementedError(u'The SQL backend does not implement mandatory '
+                                      u'traversals inside an @optional scope.')
+
+        # Sanitize literal columns to be used in the query
+        depth_string = str(depth)
+        if any(not char.isdigit() for char in depth_string):
```

Comment:
`depth` here is supposed to be a positive integer, right? Any reason to sanitize its `str` representation instead of checking the `int` value directly?

Reply:
Just being paranoid. I can subclass `int` and override the `to_string` method to return some sql injection string.

Reply:
I think it might be unnecessary. The `depth` parameter here is not user-provided, and would come from the AST of the `@recurse` directive, where that value is a raw Python `int` object. If you can define subclasses of `int` and interpose on function calls between compiler internals, then you have already gained so many capabilities that I doubt this will be even so much as a speed bump.

I'd just do a type check for `int`, assert it's a positive number, and call it a day :)